src/core/permissions/utils/computePermissionAssignments.js

```diff
@@ -47,7 +50,9 @@ function computePermissionAssignments(
     }
   });
 
-  return { isAssignedToAll, isAssignedToOnlyNonPII };
+  const isOwnerOnAtLeastOneColumn = ownershipByColumn.reduce((result, isOwner) => isOwner, true);
```

Comment:
This doesn't work. All this does is return the last value in the array. Separately, you don't need to do this if you switch to my suggestion above.